marconi-core/test/Marconi/CoreSpec.hs

```diff
@@ -1633,3 +1635,106 @@ withRollbackFailureTest =
   Tasty.testProperty "Rollback failure in a worker exit nicely" $
     Test.withMaxSuccess 10_000 $
       checkOutOfRollback (Lens.view defaultChain <$> Test.arbitrary)
+
+-- * Query modification - Stability
+withStabilityTestGroup :: Tasty.TestTree
+withStabilityTestGroup =
+  Tasty.testGroup
+    "StabilityTests"
+    [ Tasty.testGroup
+        "Stability"
+        [ Tasty.testGroup
+            "stability tests"
+            [ Tasty.testProperty "all results are stable" allStable
+            , Tasty.testProperty "not all results are stable" notAllStable
+            , Tasty.testProperty "withStability will not alter the events" withStabilityUnchanged
+            , Tasty.testProperty "no results are stable at a specific point" notStableAt
+            , Tasty.testProperty "all results are stable at a specific point" stableAt
+            , Tasty.testProperty "withStabilityAt will not alter the events" withStabilityAtUnchanged
+            ]
+        ]
+    ]
+
+-- | Helper for common chain generation logic
+genCommonChain :: (Int -> TestPoint) -> Gen ([Item TestEvent], Int, Word, Int)
```

Comment:
As a general rule, we try to avoid tuples, we prefer creating a dedicated data structure with named fields, it improves readability and maintainability (knowing the purpose of each field is challenging).